NumPy.py

```diff
@@ -1,12 +1,236 @@
 import numpy as np
+import matplotlib.pyplot as plt
 
 
 # ----------q1----------
-def change_sign():
-    array = np.array(range(21))
-    np.copysign(array[9:16], -1)
-    return array
+def question1():
+    """
+    The function creates a vector with values from 0 to 20 and change the sign of the numbers in the range from 9 to 15.
+    :return: The vector after the change.
+    """
+    vector = np.array(range(21))
+    vector[9:16] *= -1
+    return vector
+
+
+# ----------q2----------
+def question2():
+    """
+    The function creates a vector of length 10 with values evenly distributed between 5 and 50.
+    :return: The vector that created.
+    """
+    return np.linspace(5, 50, 10)
+
+
+# ----------q3----------
+def question3(matrix):
+    """
+    The function finds number of rows and columns of a given matrix.
+    :param matrix: The given matrix.
+    :return: The matrix after the change.
+    """
+    return np.array(matrix).shape
+
+
+# ----------q4----------
+def question4():
+    """
+    The function creates a 10x10 matrix, in which the elements on the borders will be equal to 1, and inside 0.
+    :return: The matrix that created.
+    """
+    matrix = np.ones((10, 10))
+    matrix[1:-1, 1:-1] = 0
+    return matrix
+
+
+# ----------q5----------
+def question5(matrix, vector):
+    """
+    The function adds the vector to each row of a given matrix.
+    :param matrix: The given matrix.
+    :param vector: The vector that adds.
+    :return: The matrix after change.
+    """
+    matrix_after_add = np.empty_like(matrix)
+    for row in range(len(matrix)):
+        matrix_after_add[row, :] = matrix[row, :] + vector
+
+    return matrix_after_add
+
+
+# ----------q6----------
+def question6(x, y):
+    """
+    The function computes the x and y coordinates for points on a sine curve and plot the points using matplotlib.
+    :param x: coordinate x.
+    :param y: coordinate y.
+    """
+    x = np.arange(x, y * np.pi, 0.1)
+    y = np.sin(x)
+    plt.plot(x, y)
+    plt.show()
+
+
+# ----------q7----------
+def question7():
+    """
+    The function creates a 4x4 array with random values, after creates a new array from the said array swapping first
+    and last rows.
+    :return: The matrix after change.
+    """
+    matrix = np.random.rand(4, 4)
+    return matrix[::-1]
+
+
+# ----------q8----------
+def question8(matrix, number):
+    """
+    The function replaces all numbers in a given array which is equal, less and greater to a given number.
+    The function prints the results.
+    :param matrix: The given matrix.
+    :param number: The number with the exchange is made.
+    """
+    number_to_replace = 0
+
+    print(np.where(matrix == number, number_to_replace, matrix))
+    print(np.where(matrix < number, number_to_replace, matrix))
+    print(np.where(matrix > number, number_to_replace, matrix))
```

Comment:
Why are you printing and not returning? You shouldn't mix logic with IO.
https://wiki.c2.com/?SeparateIoFromCalculation